docs/rules/prefer-map-has.md

```diff
@@ -0,0 +1,52 @@
+# Prefer `Map#has` over `Map#get`
+
+💼 This rule is enabled in the ✅ `recommended` [config](https://github.com/sindresorhus/eslint-plugin-unicorn#preset-configs-eslintconfigjs).
+
+💡 This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).
+
+<!-- end auto-generated rule header -->
+<!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
+
+<!-- Remove this comment, add more detailed description. -->
+
+## Fail
+
+```js
+if (map.get(foo));
+if (map.get(foo) || otherCondition);
+if (condition) else if (map.get(foo)) {};
+map.get(foo) ? 1 : 2;
+
+Boolean(map.get(foo));
+new Boolean(map.get(foo));
+
+!map.get(foo);
+!!map.get(foo);
+
+while (map.get(foo)) {
+  // do staff
+}
+
+do {} while (map.get(foo));
+
+do {} while (map.get(foo) || otherCondition);
+```
+
+## Pass
+
+```js
+if (map.has(foo));
+if (map.has(foo) || otherCondition);
+if (condition) else if (map.has(foo)) {};
+map.has(foo) ? 1 : 2;
+
+map.has(foo);
+
+while (map.has(foo)) {
+  // do staff
+}
+
+do {} while (map.has(foo));
+
+do {} while (map.has(foo) || otherCondition);
+```
```

Comment:
This is a good example, on why we should start doing this: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1751

Please format it like in https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1751#issuecomment-1254913032, with one code block per example.

Reply:
Maybe I misunderstood. According to this format, there are no longer two code blocks `Fail` and `Pass`.

Has it become two code blocks: `Blocks` and `Single-line`?

Reply:
I did update a version of this